### PR TITLE
Search: Configurable display options for the Cards result view

### PR DESCRIPTION
**Pull request by telsch**
Add options to display Keywords and Notes in Card View.
May related to #1164

Acceptance Criteria:

- [x] **Features and enhancements are fully implemented** so that they can be released at any time without additional work
- [ ] **Automated unit and/or acceptance tests have been added** to ensure the changes work as expected and to reduce repetitive manual work
- [x] **User interface changes are fully responsive** and have been tested on all major browsers and various devices
- [ ] Database-related changes are compatible with SQLite and MariaDB
- [x] Translations have been / will be updated (specify if needed)
       'Show Keywords', 'Show Notes', 'Card View Display options'
- [ ] Documentation has been / will be updated (specify if needed)
- [x] Contributor License Agreement (CLA) has been signed

### CODE REVIEW

**Review by heikomat — CHANGES_REQUESTED** [review]
Cards (amongst other things) are pseudo-virtualized.
That means that cards that are far enough outside the viewport only render partially to improve overall performance.

For this to work, the pseudo-virtualized cards need to have the same outer size as their regular in-view counterparts. If they don't, the position of the cards might start to jump around when scrolling.

Because dynamic texts inside the cards (like the keywords and notes) will affect the overall size of the card in an unpredictable way, these texts need to also be rendered in the virtualized cards (so they maintain their exact size).

You'll notice, that all texts inside the cards are rendered in two places in the `cards.vue`.
Once for the virtualized card, once for the regular in-view card.
Please make sure, that your newly added texts are also rendered in the virtualized cards.
They need to be added at about line 68, just below the `{{ photo.locationInfo() }}`-block

**Comment by telsch** [comment]
Added the newly texts also in the rendered virtualized cards and fixed the album view.

**Comment by lastzero** [comment]
@heikomat Does the frontend part look good to you now? Looking at the backend code, especially the SQL joins, I'm afraid this may have performance implications. We intentionally put all the large text fields in the details table so they wouldn't get in the way when searching. If we now always join with this table, that advantage would seem to be lost. I think I also spotted a space that might break the query (see comment).

File: internal/search/photos.go [hunk 1]
```diff
@@ -59,7 +59,8 @@ func searchPhotos(f form.SearchPhotos, sess *entity.Session, resultCols string)
 		Joins("JOIN photos ON files.photo_id = photos.id AND files.media_id IS NOT NULL").
 		Joins("LEFT JOIN cameras ON photos.camera_id = cameras.id").
 		Joins("LEFT JOIN lenses ON photos.lens_id = lenses.id").
-		Joins("LEFT JOIN places ON photos.place_id = places.id")
+		Joins("LEFT JOIN places ON photos.place_id = places.id").
+		Joins("LEFT JOIN details ON details .photo_id = photos.id")
```
Comment (lastzero) [inline internal/search/photos.go:63]: Probably should be `Joins("LEFT JOIN details ON details.photo_id = photos.id")`, so without the extra space?

**Comment by lastzero** [comment]
Since we can't easily merge it given the potential issues and questions outlined above, I'm closing this PR for now. However, we plan to add similar functionality once we have time to work on a concept and do the necessary testing to avoid bugs in a core functionality that would impact all of our users.

@telsch If you have time to get the changes to the point where we can merge them without further refactoring and testing, feel free to reopen the PR or create a new one. Thank you very much for your contribution!

**Pull request closed without merging**